### DISCUSSION TITLE
js files updated on compilation errors

### DISCUSSION THREAD

**Discussion by joprice** [fable-compiler/Fable#3848] [Q&A]
While running fable in watch mode alongside Vite, I noticed that my app was restarting or hot reloading in cases where the fsharp code had either a syntax or type error. At first I thought this might be a misconfigured watcher, but checking the output, I do see that the js files are being generated in a way where there are "holes" with `throw 1`, `defaultOf()` or similar where the error occurred. I'm sure it's a large change and maybe not feasible, but I'm curious if it's possible to instead retain the previous compilation result, only generating or modifying files on successful compilation.

**Comment by joprice** [comment]
I hit another issue which I think is related to this one: on first compile, if there are errors, the process fails with an non-zero exit code and shows the errors. If the command is run again, `Skipped compilation because all generated files are up-to-date!` in printed and the command succeeds. I assume this is because it is doing a "shallow" check for the existence of output files and not maintaining some sort of build state to indicate that the file failed on last run and still needs to be built.

**Reply by MangelMaxime** [reply]
To workaround this issue, you can pass `--noCache` so it cleans the cache on the first run.

**Comment by MangelMaxime** [comment]
I am not sure what the action to take from here is. Because, we will need to check how it works from a Developer Experience and what is best.

I believe the file compilation is done in parallels and there writing too. If we don't write `throw 1` etc. to notify that Fable compilation fails, we could perhaps end up, with `FileA` being updated but not `FileB`. However, because `FileB` is not modified with a `throw 1`, it could be possible that the applications seems to works because the old compilation result is somewhat compatible with the new `FileA`.

My scenario above is hypothetical, but worth checking and thinking about I believe.

From a personal experience, having hot reloading happening on a failed compilation never really posed problems to me. Because, once the application come back in a successful compilation state then the application state is restored. This is especially, true with Elmish.HMR in general. Perhaps with, pure React application it is a little bit different but it never bothered me a lot.
